tools/src/wyvern/target/corewyvernIL/expression/FieldSet.java

```diff
@@ -45,15 +45,30 @@ public IExpr getExprToAssign() {
 		return exprToAssign;
 	}
 
+	private boolean settingDynamicObject(TypeContext ctx) {
+		ValueType receiverType = null;
+		if (objectExpr instanceof FieldGet) {
+			IExpr receiver = ((FieldGet)objectExpr).getObjectExpr();
+			receiverType = receiver.typeCheck(ctx);
+		}
+		else if (objectExpr instanceof Variable) {
+			receiverType = ctx.lookupTypeOf(((Variable)objectExpr).getName());
+		}
+		else throw new RuntimeException("Target of FieldSet is unsupported. Type: " + objectExpr.getClass());
+		return Util.isDynamicType(receiverType);
+	}
+	
 	@Override
 	public ValueType typeCheck(TypeContext ctx) {
 
+		if (settingDynamicObject(ctx)) return Util.dynType();
```

Comment:
Should make sure the right-hand side has a type (it can be any time, but it can't have a type error).